pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -324,6 +327,26 @@ type ACMEIssuerDNS01ProviderRFC2136 struct {
 	TSIGAlgorithm string `json:"tsigAlgorithm,omitempty"`
 }
 
+// ACMEIssuerDNS01ProviderWebhook is a structure containing the
+// configuration for Webhook DNS
+type ACMEIssuerDNS01ProviderWebhook struct {
+	// URL of a webhook
+	URL string `json:"url"`
+
+	// Provide arbitrary k/v metadata in a request to webhook
+	// +optional
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// Skip verification of a TLS certificate presented by webhook
+	// +optional
+	SkipTLSVerify bool `json:"skipTLSVerify"`
+
+	// Certificate Authority used to verify webook certificate
+	// Required if "skipTLSVerify" is "false"
+	// +optional
+	CaSecret SecretKeySelector `json:"CaSecretRef"`
```

Comment:
This doesn't have to be a secret, since it's just a root cert, not a serving key/cert pair

Reply:
I'd argue for my approach for the following reasons:
1. There is already a validating machinery for Secrets;
2. It practice there aren't a lot of difference between Secrets and ConfigMaps. Some people might think that information warrants being stored in a secret. Why deny them?